libs/gretty/src/main/groovy/org/akhikhl/gretty/ProjectUtils.groovy

```diff
@@ -84,20 +84,37 @@ final class ProjectUtils {
   }
 
   private static void addProjectClassPathWithDependencies(Set<URL> urls, Project proj, String dependencyConfigName) {
-    if(proj.sourceSets.main.output.files.any { new File(it, 'META-INF/web-fragment.xml').exists() })
-      urls.add proj.jar.archivePath.toURI().toURL()
-    else
-      urls.addAll proj.sourceSets.main.output.files.collect { it.toURI().toURL() }
     def dependencyConfig = proj.configurations.findByName(dependencyConfigName)
     if(dependencyConfig) {
-      def files = dependencyConfig.fileCollection { !(it instanceof ProjectDependency) }
+      Set<File> dependencyProjectArchives = new LinkedHashSet<>()
+
+      def collectProjectDependencies
+      collectProjectDependencies= { Project project ->
+        def outputFiles = project.sourceSets.main.output.files
+        if(outputFiles.any { new File(it, 'META-INF/web-fragment.xml').exists() }) {
+          urls.add(project.jar.archivePath.toURI().toURL())
+        } else {
+          urls.addAll(outputFiles.collect { it.toURI().toURL() })
+          dependencyProjectArchives.add(project.jar.archivePath)
```

Comment:
I just realized [`project.jar.archivePath`](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.bundling.Jar.html#org.gradle.api.tasks.bundling.Jar:archivePath) is already deprecated. I'd be glad if we could fix this within the scope of this PR.

Reply:
@yuezk thank you for the fix of the deprecation 💯